handsontable/src/plugins/filters/filters.js

```diff
@@ -134,6 +134,13 @@ export class Filters extends BasePlugin {
    * @type {WeakSet<Menu>}
    */
   #dropdownMenuTraces = new WeakSet();
+  /**
+   * Stores the previous state of the condition stack before the latest filter operation.
+   * This is used in the `beforeFilter` plugin to allow performing the undo operation.
+   *
+   * @type {Array}
+   */
+  #previousConditionStack = [];
```

Comment:
is there any undo/redo architecture documentation ?

Reply:
Currently, we don't have any architecture for undo/redo states. Usually each plugin implements it in a different way.